### ISSUE TITLE
MIDI importer

### ISSUE THREAD

**Issue by mtolly** [iimarckus/pokered#49]
After coming across this wonderful project, I saw [an issue in the Crystal disassembly planning a MIDI importer](https://github.com/kanzure/pokecrystal/issues/58), and after studying the music macro format a bit I decided to bang one out for Red over the last week:

https://github.com/mtolly/pokemid

Currently the main features are: it takes care of turning MIDI durations and pitches into the right sequence of note speed and octave commands, it supports both one-shot and looping tracks, as well as most of the note alteration macros, and also does some rudimentary code shrinking by finding repeated sections and making subroutines.

For now it is a ~500-line Haskell program, but I've put up [binaries on the release page](https://github.com/mtolly/pokemid/releases) for Windows/Mac/Linux so you hopefully shouldn't need to install anything to try it out. It comes with 2 short MIDI files as tests, which generate equivalent (but not identical) code for `introbattle.asm` and `safarizone.asm`.

Basically I'd like to know what you guys want out of the tool. It is not finished but all the core logic and design is there. The primary issue is that it does not generate identical code to the original files, only code which behaves equivalently (to my knowledge). Emitting identical assembly can't be easily done without making the MIDI files specify more low-level details of labels and branching which doesn't seem useful to me. So I wouldn't really recommend using this to store the repo's music in MIDI like how the images are stored in PNG.

**Comment by mtolly** [comment]
Gonna close this issue since I don't think I'll have any more questions about the music format at this point. Anyone having issues with the program please hit up the issue tracker I linked above!
